TopMovies/Service/MovieService.swift

```diff
@@ -9,29 +9,34 @@ import Foundation
 import ReSwift
 
 class MovieService: StoreSubscriber {
-    typealias StoreSubscriberStateType = MainState
-    
-    private let movieAPI: MovieAPIProtocol
-    
-    init(movieAPI: MovieAPIProtocol) {
-        self.movieAPI = movieAPI
-        mainStore.subscribe(self)
-    }
-    
-    func newState(state: MainState) {
-        switch state.moviesListState.moviesList {
-        case .requested:
-            mainStore.dispatch(MoviesListAction.downloading)
-            movieAPI.topMovies { (result) in
-                switch result {
-                case let .success(moviesList):
-                    mainStore.dispatch(MoviesListAction
-                                        .completed(movies: moviesList.results.map { Movie(dto: $0) }))
-                case let .failure(error):
-                    mainStore.dispatch(MoviesListAction.failed(error: error))
-                }
-            }
-        default: break
+  typealias StoreSubscriberStateType = MainState
+  
+  private let movieAPI: MovieAPIProtocol
+  
+  init(movieAPI: MovieAPIProtocol) {
+    self.movieAPI = movieAPI
+    mainStore.subscribe(self)
+  }
+  
+  func newState(state: MainState) {
+    switch state.movieCategoriesState.categoriesList {
+    case .requested:
+      mainStore.dispatch(MovieCategoriesAction.downloading)
+      movieAPI.topMovies { (result) in
+        switch result {
+        case let .success(categoriesList):
+          mainStore.dispatch(MovieCategoriesAction
+                              .completed(categories: categoriesList.map{ MovieCategory(dto: $0) }))
+          mainStore.dispatch(MoviesDownloadingAction
+                              .completed(
+                                MovieCategory(dto: categoriesList.first!).id,
+                                categoriesList.first!.results.map{ Movie(dto: $0) }
```

Comment:
It is unsafe to use force unwrap. Try to avoid that. 
I feel that you're planning to rewrite that in a single action (I hope so :)), but try to instill a habit to avoid force unwrap.
For example:  `categoriesList.first?.results.map{ Movie(dto: $0) ?? []` and so on.
Moreover, you don't handle the case when API returns you an empty array of categories ;)